pyomo/contrib/viewer/pyomo_viewer.py

```diff
@@ -22,24 +22,15 @@
 
 import os
 
+from pyomo.common.dependencies import attempt_import
 from pyomo.scripting.pyomo_parser import add_subparser
 import pyomo.contrib.viewer.qt as myqt
 
-qtconsole_available = False
-qtconsole_errors = list(myqt.import_errors)
-if myqt.available:
-    try:
-        from qtpy import QtGui, QtCore, QtWidgets
-        import qtconsole.qtconsoleapp as qtconsole_app
-
-        qtconsole_available = True
-    except ImportError as e:
-        qtconsole_errors.append(str(e))
+qtconsole_app, qtconsole_available = attempt_import('qtconsole.qtconsoleapp')
 
-if qtconsole_available:
 
-    class QtApp(qtconsole_app.JupyterQtConsoleApp):
-        _kernel_cmd_show_ui = """try:
+class QtApp(qtconsole_app.JupyterQtConsoleApp if qtconsole_available else object):
```

Comment:
Same question here. Inheriting from object could cause some strange errors if qtconsole is not available. You won't necessarily get the deferred import errors???

Reply:
Same solution as above.